### ISSUE TITLE
Difference between VEP htslib and regular htslib

### ISSUE THREAD

**Issue by Stikus** [Ensembl/ensembl-vep#447]
Hello.
We using VEP in Docker-container and we have samtools installed in same container.
Thats the reason why we have 2 different copies of htslib in container.
VEP INSTALL.pl has flag to suppress htslib installation (--NO_HTSLIB), but according to this parl of INSTALL.pl VEP installation modifying default htslib:
https://github.com/Ensembl/ensembl-vep/blob/release/96/INSTALL.pl#L843
```
    if (/^CFLAGS/ && !/-fPIC/) {
      s/#.+//;  # get rid of comments
      $_ .= " -fPIC -Wno-unused -Wno-unused-result";
    }
```
Is this part still needed and if yes - for what reason? Or we can use htslib from samtools without fear?

**Comment by aparton** [comment]
Hiya,

We explicitly download version 1.3.2 of HTSlib for use with VEP because newer versions have dependancies on `lzma` and `bz2` that we didn't want to introduce into VEP. As long as you're happy with these dependancies, I don't think there should be a problem, although we haven't done any rigorous testing with the latest version of HTSlib.

Regarding the installation flags, -Wno-unused -Wno-unused-result are simply suppressing warnings and -fPIC tells gcc to produce position-independent code. Depending on how samtools was installed and setup, you may need to be wary of -fPIC, but I don't expect it to be an issue.

In short, you should be fine to use the version of htslib built into samtools, although we haven't done any extensive testing with it. Please let us know if you find any issues with it.

Kind Regards,
Andrew

**Comment by serge2016** [comment]
@aparton, and what is wrong with dependencies on `lzma` and `bz2`? Do they influence the efficiency of VEP?

**Comment by aparton** [comment]
Hi,

Introducing these dependancies wouldn’t slow down VEP, the decision was made solely to prevent installation complications. Additionally, lzma was tricky to install on Mac OS, so we decided to stick with an older version of HTSlib.

We’ve had some conversations today about upgrading the version used by VEP, and this is something that we’re going to look into in the coming weeks.

Kind Regards,
Andrew

**Comment by serge2016** [comment]
Andrew, thank you for the answer! And for the work!
